json_serialization/writer.nim

```diff
@@ -365,8 +365,12 @@ proc writeValue*(w: var JsonWriter, value: auto) {.gcsafe, raises: [IOError].} =
     #      to avoid the allocation here:
     append $value
 
-  elif value is (seq or array or openArray):
-    w.writeArray(value)
+  elif value is (seq or array or openArray) or
+      (value is distinct and distinctBase(value) is (seq or array or openArray)):
+    when value is distinct:
+      w.writeArray(distinctBase value)
```

Comment:
Doesn't that ignore an override of `writeValue` if we have:

- A = seq[int] (base)
- B = distinct A
- C = distinct B

if we write `C` and have a custom `writeValue` on `B`?

`distinctBase` by default applies recursively, so `distinctBase(C)` is `A` not `B`.